### PR TITLE
[needs attention] - frontend/feature: add guest menu on landing page, refactor cookie banner

### LINKED ISSUE #4127: Couchers.org landing page improvements
The following should be done on the couchers.org landing page (to clarify, this is landing on the page as a guest/anonymous/before logging on). 

Edit landing page / header
- [ ] Remove the report a problem, sign up, login button from the header. 
- [ ] Add anonymous/guest profile drop down in top right header (as seen in the header when logged on as a user). Dropdown menu should have a list of features that are accessible as a guest (sign up, log in, help, report a problem).

Edit cookie disclosure design 
- [ ] Expand the frame to fit width screen
- [ ] Constrain frame to bottom the page
- [ ] Change text color to black
- [ ] Change link color to a primary green
- [ ] Center align the text
- [ ] Remove the "X" component 

I can provide screenshots below. Message me if you have any questions or would like to read my formal write up on these!

### PR DESCRIPTION
**Pull request by marco-digennaro**
- It adds a guest menu on navbar with links to sign-in, sign-up, report bug, help
- it refactors the cookie banner fixing it at bottom

About last point check:
https://github.com/Couchers-org/couchers/issues/4127#issuecomment-2439445903

Closes #4127

**Backend checklist**
- [ ] Formatted my code by running `ruff check --select I --fix . && ruff check . && ruff format .` in `app/backend`
- [ ] Added tests for any new code or added a regression test if fixing a bug
- [ ] All tests pass
- [ ] Run the backend locally and it works
- [ ] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

**Web frontend checklist**
- [ x] Formatted my code with `yarn format`
- [ x] There are no warnings from `yarn lint --fix`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ x] All tests pass
- [x ] Clicked around my changes running locally and it works
- [ x] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by nabramow** [comment]
@marco-digennaro Are you still working on this or are you ready for review? I'm unclear as it's marked "ready for review" but the title says [needs attention].

**Comment by marco-digennaro** [comment]
> @marco-digennaro Are you still working on this or are you ready for review? I'm unclear as it's marked "ready for review" but the title says [needs attention].

it's actually done, i've made all the changes requested in the ticket, but i have a doubt that i wrote in a comment of the ticket, the cookie banner covers the lower footer and without the close button (as requested) cannot be closed. should it be that way?

**Comment by bakeiro** [comment]
what I see in this PR (reason why I wanted to discuss the report button in the meeting) it's that the report button it's removed from navbar, and only accesible only after log in, I wouldn't do it this way, since some people report sign up/log in problems, and I believe it's useful to track that, I agree makes more sense from the UX perspective to have it only after log in, but we should think about the users who sometimes have these login/sign up problems

**Comment by bakeiro** [comment]
@marco-digennaro maybe can we keep the button until is discussed later on what to do with it?

**Comment by bakeiro** [comment]
also I don't understand why removing the 'x' from the cookie banner... we can chat about this tomorrow in the meeting.  
note that Dom, who created the ticket (AFAIK) isn't active in the project, so I don't think he can answer our doubts/questions about his tickets... but we have other tickets created ourselves instead which we know how to address them

**Comment by marco-digennaro** [comment]
> what I see in this PR (reason why I wanted to discuss the report button in the meeting) it's that the report button it's removed from navbar, and only accesible only after log in, I wouldn't do it this way, since some people report sign up/log in problems, and I believe it's useful to track that, I agree makes more sense from the UX perspective to have it only after log in, but we should think about the users who sometimes have these login/sign up problems

well actually no, the report button is removed from the bar and added in the guest menu, so still accessible if not logged

**Comment by marco-digennaro** [comment]
> also I don't understand why removing the 'x' from the cookie banner... we can chat about this tomorrow in the meeting. note that Dom, who created the ticket (AFAIK) isn't active in the project, so I don't think he can answer our doubts/questions about his tickets... but we have other tickets created ourselves instead which we know how to address them

that was my question to the whole ticket: should the close button be removed? that way the banner keeps covering the lower footer. so i guess the close button should stay

**Comment by bakeiro** [comment]
> > what I see in this PR (reason why I wanted to discuss the report button in the meeting) it's that the report button it's removed from navbar, and only accesible only after log in, I wouldn't do it this way, since some people report sign up/log in problems, and I believe it's useful to track that, I agree makes more sense from the UX perspective to have it only after log in, but we should think about the users who sometimes have these login/sign up problems
> 
> well actually no, the report button is removed from the bar and added in the guest menu, so still accessible if not logged

oh sorry! I misunderstood then... all good!

**Comment by bakeiro** [comment]
> > also I don't understand why removing the 'x' from the cookie banner... we can chat about this tomorrow in the meeting. note that Dom, who created the ticket (AFAIK) isn't active in the project, so I don't think he can answer our doubts/questions about his tickets... but we have other tickets created ourselves instead which we know how to address them
> 
> that was my question to the whole ticket: should the close button be removed? that way the banner keeps covering the lower footer. so i guess the close button should stay

ok, don't worry, I'll ask this in the meeting and I'll let U know our answer, sorry for the delay 😞

**Comment by bakeiro** [comment]
Hi!

we decided to keep the 'x', not sure why the ticket says to delete it... sorry for the confusion!!

**Comment by marco-digennaro** [comment]
> Hi!
> 
> we decided to keep the 'x', not sure why the ticket says to delete it... sorry for the confusion!!

good!
i've updated the PR!

File: app/web/components/CookieBanner.tsx [hunk 1]
```diff
@@ -11,46 +11,56 @@ import makeStyles from "utils/makeStyles";
...
+    transform: 'translateY(-50%)',
+    right: theme.spacing(1),
+  }
 }));
 
 export default function CookieBanner() {
   const { t } = useTranslation();
   const classes = useStyles();
-  const [hasSeen, setHasSeen] = usePersistedState("hasSeenCookieBanner", false);
+  // const [hasSeen, setHasSeen] = usePersistedState("hasSeenCookieBanner", false);
```
Comment (bakeiro) [inline app/web/components/CookieBanner.tsx]: delete?

**Review by bakeiro — DISMISSED** [review]

**Comment by bakeiro** [comment]
thanks for your time and sorry for the confusion :)

**Comment by bakeiro** [comment]
before merging

- [x] was this checked? [if so proceed to merge :)]
image

**Comment by nabramow** [comment]
Does this also address #4128 ? Trying to decide if I can close that one!

**Review by bakeiro — APPROVED** [review]

**Pull request merged**